### DISCUSSION TITLE
RemoteTransform could potentially offer an array of remote paths

### DISCUSSION THREAD

**Discussion by amonroejj** [godotengine/godot-proposals#4356] [3D]
RemoteTransform could potentially offer an array of remote paths. Currently it only offers the option of a single remote path.

Use case for this: Synchronizing the position of multiple cameras in multiple viewports. A side effect of Godot's unfriendly ViewportContainer/Viewport/Camera design is that the Cameras are "children but not really children" of a parent game object. They appear as children in the scene hierarchy but do not respect the user's expectations in following the position of the parent game object.

See https://youtu.be/uPaEx35bWSA?t=237 for gameplay from the old Moraff's World dungeon crawler game that uses four linked cameras pointing North, South, East and West.

This can be worked around by script pretty easily, but the RemoteTransform array would be more beginner-friendly.

**Comment by Calinou** [comment]
> This can be worked around by script pretty easily

Since this doesn't seem like a very common use case to me, I think it's better to leave this to script/add-on territory.

Feel free to publish an add-on to provide this functionality :slightly_smiling_face:

**Reply by wareya** [reply]
> Since this doesn't seem like a very common use case to me, I think it's better to leave this to script/add-on territory.

This seems like a cop-out. The whole point of RemoteTransform is to reduce the developer's reliance on scripts when working around the limitations of godot's node hierarchy model.

This proposal idea applies to more than just cameras. I have a game with turn-based movement where the parent's position is rounded to tiles, but sprites, floating health bars, labels, etc. have to follow an interpolated position during turn transitions. Without a RemoteTransform that can target multiple remote objects at once, I need to do one of:

1) manually update the position of each visual child of the parent object in code

2) make a new RemoteTransform node for every single visual child

3) stuff every visual child into a dummy helper Node2D child called something like "Relative" and interpolate that; but now whenever I have to access any of those children in code, I have to add this dummy node to the start of its path, even though it's just doing the work that RemoteTransform should be capable of doing

**Comment by Xrayez** [comment]
> Use case for this: Synchronizing the position of multiple cameras in multiple viewports. A side effect of Godot's unfriendly ViewportContainer/Viewport/Camera design is that the Cameras are "children but not really children" of a parent game object.

This smells like the Node system limitation to be honest. See #3915.
